tasks/update_go.py

```diff
@@ -0,0 +1,179 @@
+import re
+from typing import Optional
+
+from invoke import Context, exceptions, task
+
+from .go import tidy_all
+from .libs.common.color import color_message
+from .modules import DEFAULT_MODULES
+from .pipeline import update_circleci_config, update_gitlab_config
+
+GO_VERSION_FILE = "./.go-version"
+
+
+# replace the given pattern with the given string in the file
+def _update_file(path: str, pattern: str, replace: str, expected_match: int = 1):
+    # newline='' keeps the file's newline character(s)
+    # meaning it keeps '\n' for most files and '\r\n' for windows specific files
+
+    with open(path, "r", newline='') as reader:
+        content = reader.read()
+
+    content, nb_match = re.subn(pattern, replace, content, flags=re.MULTILINE)
+    if nb_match != expected_match:
+        print(
+            color_message(
+                f"WARNING: {path}: '{pattern}': expected {expected_match} matches but go {nb_match}", "orange"
+            )
+        )
```

Comment:
My thought is that the most likely case for this warning is that the file was changed, but I tried to make the regex precise enough so that if there is a match it's really likely a good match (we don't want to match by accident), and having the task do part of the changes is better than having it do nothing.

On the other hand if the task does nothing it would likely push people to fix it, so it might be a good choice.
Do you have a preference ?

Reply:
Changed the behavior to exit by default in case of mismatch. Added a `warn` flag to change that to just print a warning instead